cli/src/revset_util.rs

```diff
@@ -77,7 +97,11 @@ impl<'repo> RevsetExpressionEvaluator<'repo> {
 
     /// Evaluates the expression.
     pub fn evaluate(&self) -> Result<Box<dyn Revset + 'repo>, UserRevsetEvaluationError> {
-        let symbol_resolver = default_symbol_resolver(self.repo, self.id_prefix_context);
+        let symbol_resolver = default_symbol_resolver(
+            self.repo,
+            &self.extensions.symbol_resolvers,
+            self.id_prefix_context,
+        );
```

Comment:
Something like `id_prefix_context.to_symbol_resolver(repo)` might be useful to get rid of `extensions` arguments. It means `IdPrefixContext` can't be initialized without `extensions`, though.

Reply:
I think this is incompatible with my clarification on the previous comment.

Reply:
> I think this is incompatible with my clarification on the previous comment.

Which comment?

My point here is that the `extensions` list could be carried by `IdPrefixContext` because the context owns the list anyway. Maybe it would go against the original design of `IdPrefixContext`, but holding `extensions` in `DisambiguationData` would be the same.

Reply:
Hmm, alright I see it now.  I think it is cleaner to store the extensions on IdPrefixContext but I wouldn't go so far as to add the `to_symbol_resolver()` method, that feels a bit heavy-handed.

Reply:
This is off-topic, but `IdPrefixContent` or its caller will need a subset of `DefaultSymbolResolver` (or `[PartialSymbolResolver]`) to ensure that the shortest hex prefix doesn't conflict with named symbols. I'm not sure how these modules interact, but there will be an object that can do both named symbol lookup and prefix resolution/disambiguation, separately.

https://github.com/martinvonz/jj/issues/2416

Reply:
I think we'll resolve this in the future with symbol resolver ordering; an extension can declare whether it should be resolved before or after id prefixes.